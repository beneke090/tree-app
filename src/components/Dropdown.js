/* eslint-disable jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';

function Dropdown({ values, label, onChange }) {
  return (
    <label className="block mt-4">
      <span className="text-gray-700">{label}</span>
      <select
        id=""
        name="select"
        className="form-select mt-1 block w-full"
        onChange={e => onChange(e.target.value)}
      >
        {values.map((item, i) => (
          <option key={item} value={item}>
            {item}
          </option>
        ))}
      </select>
    </label>
  );
}

Dropdown.propTypes = {
  label: PropTypes.string.isRequired,
  values: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Dropdown;
