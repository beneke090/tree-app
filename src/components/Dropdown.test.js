import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';

describe('Dropdown', () => {
  test('should match snapshot.', () => {
    const component = renderer.create(
      <Dropdown
        label="testlabel"
        values={['a', 'b', 'c']}
        onChange={() => {}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
