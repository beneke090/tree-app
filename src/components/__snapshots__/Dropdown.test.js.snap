// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match snapshot. 1`] = `
<label
  className="block mt-4"
>
  <span
    className="text-gray-700"
  >
    testlabel
  </span>
  <select
    className="form-select mt-1 block w-full"
    id=""
    name="select"
    onChange={[Function]}
  >
    <option
      value="a"
    >
      a
    </option>
    <option
      value="b"
    >
      b
    </option>
    <option
      value="c"
    >
      c
    </option>
  </select>
</label>
`;
